runtime/src/rent_collector.rs

```diff
@@ -0,0 +1,90 @@
+//! Bank's wrapper around `RentCollector` to allow for overriding of some
+//! `SVMRentCollector` trait methods, which are otherwise implemented on
+//! `RentCollector` directly.
+//!
+//! Agave requires submission of logs and metrics during account rent state
+//! assessment, which is not included in the `RentCollector` implementation
+//! of the `SVMRentCollector` trait. This wrapper allows all `SVMRentCollector`
+//! methods to be passed through to the underlying `RentCollector`, except for
+//! those which require additional logging and metrics.
+
+use {
+    log::*,
+    solana_sdk::{
+        account::AccountSharedData,
+        clock::Epoch,
+        pubkey::Pubkey,
+        rent::{Rent, RentDue},
+        rent_collector::{CollectedInfo, RentCollector},
+        transaction::{Result, TransactionError},
+        transaction_context::IndexOfAccount,
+    },
+    solana_svm_rent_collector::{rent_state::RentState, svm_rent_collector::SVMRentCollector},
+};
+
+/// Wrapper around `RentCollector` to allow for overriding of some
+/// `SVMRentCollector` trait methods, which are otherwise implemented on
+/// `RentCollector` directly.
+///
+/// Overrides inject logging and metrics submission into the rent state
+/// assessment process.
+pub struct RentCollectorWithMetrics(RentCollector);
+
+impl RentCollectorWithMetrics {
+    pub fn new(rent_collector: RentCollector) -> Self {
+        Self(rent_collector)
+    }
+}
+
+impl SVMRentCollector for RentCollectorWithMetrics {
```

Comment:
Instead of having the wrapper struct, is implemented `SVMRentCollector` directly for `RentCollector` possible?

Reply:
It's already implemented from the last PR in the `svm-rent-collector` crate. We can't implement it again in `runtime` and we can't implement traits from other crates on types from other crates.